lib/Factory/EncDecAllFactory.php

```diff
@@ -0,0 +1,155 @@
+ *
+ */
+
+namespace OCA\Encryption\Factory;
+
+use OC\Encryption\DecryptAll;
+use OC\Encryption\Manager;
+use OC\Files\View;
+use OCA\Encryption\Crypto\Crypt;
+use OCA\Encryption\Crypto\CryptHSM;
+use OCA\Encryption\Crypto\EncryptAll;
+use OCA\Encryption\KeyManager;
+use OCA\Encryption\Session;
+use OCA\Encryption\Users\Setup;
+use OCA\Encryption\Util;
+use OCP\Encryption\Keys\IStorage;
+use OCP\IConfig;
+use OCP\IL10N;
+use OCP\ILogger;
+use OCP\IUserManager;
+use OCP\IUserSession;
+use OCP\Mail\IMailer;
+use OCP\Security\ISecureRandom;
+use Symfony\Component\Console\Helper\QuestionHelper;
+
+class EncDecAllFactory {
+	/** @var Manager  */
+	private $encryptionManager;
+	/** @var IUserManager  */
+	private $userManager;
+	/** @var View  */
+	private $rootView;
+	/** @var ILogger  */
+	private $logger;
+	/** @var Util  */
+	private $encUtil;
+	/** @var IConfig  */
+	private $config;
+	/** @var IMailer  */
+	private $mailer;
+	/** @var IL10N  */
+	private $l10n;
+	/** @var QuestionHelper  */
+	private $questionHelper;
+	/** @var ISecureRandom  */
+	private $secureRandom;
+	/** @var IStorage  */
+	private $encStorage;
+	/** @var IUserSession  */
+	private $userSession;
+	/** @var Session  */
+	private $encSession;
+	/** @var CryptHSM  */
+	private $cryptHSM;
+	/** @var Crypt  */
+	private $crypt;
+
+	/**
+	 * EncDecAllFactory constructor.
+	 *
+	 * @param Manager $encryptionManager
+	 * @param IUserManager $userManager
+	 * @param View $rootView
+	 * @param ILogger $logger
+	 * @param Util $encUtil
+	 * @param IConfig $config
+	 * @param IMailer $mailer
+	 * @param IL10N $l10n
+	 * @param QuestionHelper $questionHelper
+	 * @param ISecureRandom $secureRandom
+	 * @param IStorage $encStorage
+	 * @param Session $encSession
+	 * @param CryptHSM $cryptHSM
+	 * @param Crypt $crypt
+	 * @param IUserSession $userSession
+	 */
+	public function __construct(Manager $encryptionManager, IUserManager $userManager,
+								View $rootView, ILogger $logger,
+								Util $encUtil, IConfig $config, IMailer $mailer, IL10N $l10n,
+								QuestionHelper $questionHelper, ISecureRandom $secureRandom,
+								IStorage $encStorage, Session $encSession, CryptHSM $cryptHSM,
+								Crypt $crypt, IUserSession $userSession) {
+		$this->encryptionManager = $encryptionManager;
+		$this->userManager = $userManager;
+		$this->rootView = $rootView;
+		$this->logger = $logger;
+		$this->encUtil = $encUtil;
+		$this->config = $config;
+		$this->mailer = $mailer;
+		$this->l10n = $l10n;
+		$this->questionHelper = $questionHelper;
+		$this->secureRandom = $secureRandom;
+		$this->encStorage = $encStorage;
+		$this->encSession = $encSession;
+		$this->cryptHSM = $cryptHSM;
+		$this->crypt = $crypt;
+		$this->userSession = $userSession;
+	}
+
+	/**
+	 * Returns DecryptAll object
+	 *
+	 * @return DecryptAll
+	 */
+	public function getDecryptAllObj() {
+		return new DecryptAll($this->encryptionManager, $this->userManager, $this->rootView, $this->logger);
```

Comment:
It makes more sense to create a new `View` object in order to ensure the `DecryptAll` object has the root view. No need to inject a view.

Reply:
Injected `View` object both in the `EncryptAll` and the `DecryptAll`.